### PR TITLE
fix/ #2565 If I've already closed the banner, never have it show on on FE reload

### PR DESCRIPTION
**Pull request by HarjobandeepSingh**
**What is the purpose of the change**
This pull request modifies the behavior of the announcement banner on the website. The change ensures that once a user closes the banner, it does not reappear upon page reload, improving the user experience.

**Brief Changelog**
Modified the AdBannerContent component to use local storage for tracking the visibility state of the banner.
Added a click event handler to the entire banner that sets the banner's visibility state in local storage and hides the banner.
Testing and Verifying
This change has been tested locally by rebuilding the website and verifying that the banner does not reappear after being closed and the page is reloaded.

**Documentation and Release Note**
This pull request introduces a change in user-facing behavior. Previously, the announcement banner would reappear every time the page was reloaded, even if the user had closed it. With this change, once the user closes the banner, it will not reappear upon page reload.

### CODE REVIEW

**Comment by HarjobandeepSingh** [comment]
fix/ 2565

**Comment by HarjobandeepSingh** [comment]
@MaxMillington Fix/

File: packages/web/components/ad-banner/ad-banner-content.tsx [hunk 1]
```diff
@@ -18,12 +18,31 @@ export const AdBannerContent: React.FC<AdBannerContentProps> = memo(
...
     gradient,
   }) => {
+    const [showBanner, setShowBanner] = useState(false);
+
+    useEffect(() => {
+      const bannerStatus = localStorage.getItem('bannerStatus');
+      if (bannerStatus !== 'closed') {
+        setShowBanner(true);
+      }
+    }, []);
```
Comment (jonator) [inline packages/web/components/ad-banner/ad-banner-content.tsx:28]: FYI we have a `useLocalstorageState` hook that may reduce some of this

File: packages/web/components/ad-banner/ad-banner-content.tsx [hunk 2]
```diff
@@ -52,4 +69,4 @@ export const AdBannerContent: React.FC<AdBannerContentProps> = memo(
       </a>
     );
   }
-);
+);
```
Comment (jonator) [inline packages/web/components/ad-banner/ad-banner-content.tsx:72]: Let's make sure we run lint fix

**Comment by MaxMillington** [comment]
@HarjobandeepSingh please address @jonator's comments, then we will send to QA to review

File: packages/web/components/ad-banner/ad-banner-content.tsx [hunk 3]
```diff
@@ -39,8 +51,6 @@ export const AdBannerContent: React.FC<AdBannerContentProps> = memo(
         />
         <div
           className={classNames("flex w-full flex-col gap-1 py-2.5")}
-          // we pass this color in directly to avoid having to manually update our tailwind safelist with arbitrary values
```
Comment (MaxMillington) [inline packages/web/components/ad-banner/ad-banner-content.tsx:42]: would you add this comment back in please?

**Review by arispech — CHANGES_REQUESTED** [review]
The  banner still shows up for a second when you are refreshing the page


https://github.com/osmosis-labs/osmosis-frontend/assets/56182049/79872457-b89a-400b-ba6b-efb2916ffdc6

**Comment by CryptoAssassin1** [comment]
Same as Aris reported the banner appear disappear in a sec.

**Comment by kamal-sutra** [comment]
@HarjobandeepSingh Also notice how the Swap window flashes/shakes when clicked on Swap button. 
https://github.com/osmosis-labs/osmosis-frontend/assets/116148606/89c9210c-3762-456a-8378-777be737b8f0

**Comment by CryptoAssassin1** [comment]
@HarjobandeepSingh banner is not loading now so cant test if its fixed or not.
file:///home/assassin/Downloads/Trade%20on%20Osmosis%20Zone%20(37).mp4

**Comment by HarjobandeepSingh** [comment]
> @HarjobandeepSingh banner is not loading after deleting cookies and site data so cant test if its fixed or not.
> 
>  Trade.on.Osmosis.Zone.37.mp4

ok i will fix this

**Comment by HarjobandeepSingh** [comment]
> banner is not loading after deleting cookies and site data so cant test if its fixed or not.

so i have to fix it or not ?

File: packages/web/components/navbar/index.tsx [hunk 4]
```diff
@@ -156,11 +156,11 @@ export const NavBar: FunctionComponent<
     // announcement banner
     const [_showBanner, setShowBanner] = useLocalStorageState(
       Announcement ? Announcement?.localStorageKey ?? "" : "",
-      true
+      null
```
Comment (MaxMillington) [inline packages/web/components/navbar/index.tsx:159]: this is just not going to show the banner ever right?

You can add a banner to test this by adding the following code to line 39 in feature-flag.ts

```
  : {
    localStorageKey: "test-key",
    enTextOrLocalizationPath: "test description",
  };
  ```

**Review by MaxMillington — CHANGES_REQUESTED** [review]
the banner doesn't show at all with the current code

**Comment by MaxMillington** [comment]
Thank you for your contribution. This issue has been completed [here](https://github.com/osmosis-labs/osmosis-frontend/pull/2654)

**Pull request closed without merging**